gtests/net/tcp/common/set_sysctls.py

```diff
@@ -1,24 +1,46 @@
-#!/usr/bin/python
+#!/usr/bin/env python3
+#
+# Copyright 2012 Google Inc. All Rights Reserved.
+
+"""Sets sysctl values and writes a file that restores them.
+
+The arguments are of the form "<proc-file>=<val>" separated by spaces.
+The program first reads the current value of the proc-file and creates
+a shell script named "/tmp/sysctl_restore_${PPID}.sh" which restores the values
+when executed. It then sets the new values
+"""
+
+__author__ = ('brakmo@google.com (Lawrence Brakmo)')
 
-import commands
 import os
+import subprocess
 import sys
 
-pppid = int(os.popen("ps -p %d -oppid=" % os.getppid()).read().strip())
-filename = '/tmp/sysctl_restore_%d.sh' % pppid
+# Sometimes shell skips fork() as an optimization.
```

Comment:
This part is different from internal packetdrill. It would be good to keep this part of packetdrill upstream and internally in sync. Is the plan to backport this commit to our internal packetdrill after it is merged upstream? ( If so, this sounds fine to me. :-) )

Reply:
Now this file is in sync with internel version :-)